### ISSUE TITLE
1.9.5 Separator line before footer is not displayed in HTML Help (CHM)

### ISSUE THREAD

**Issue by parfant** [doxygen/doxygen#9546]
I have updated to doxygen 1.9.5 today. Compiling the HTML help worked without any problems.
However, as soon as I use a custom stylesheet via HTML_EXTRA_STYLESHEET, HTML Help no longer shows the separator line before the footer. In the HTML files, the separator line is displayed in the browser without any problems. For the custom stylesheet, I used the original file from doxygen 1.9.5 without any changes.

Browser:
![Browser](https://user-images.githubusercontent.com/26295766/187146643-896e5ec7-2fc3-4eeb-9a68-141be494e85f.png)

HTML Help:
![CHM](https://user-images.githubusercontent.com/26295766/187146721-ef391bf7-1f94-4dd7-a80d-f5cc0b96baf7.png)

Source:
[DoxygenTest.zip](https://github.com/doxygen/doxygen/files/9442439/DoxygenTest.zip)

**Comment by albert-github** [comment]
- You mentioned:
   >  For the custom stylesheet, I used the original file from doxygen 1.9.5 without any changes.

  How did you generate the `customdoxygen.css`?
- I assume you are running the released 1.9.5 version but to be sure can you give the full information provided by `doxygen -v`
- Which platform are you running on, what is

**Comment by parfant** [comment]
> How did you generate the `customdoxygen.css`?

I generated the customdoxygen.css in the command line with `doxygen -w html header.html footer.html customdoxygen.css`

> I assume you are running the released 1.9.5 version but to be sure can you give the full information provided by `doxygen -v`

`1.9.5 (2f6875a5ca481a69a6f32650c77a667f87d25e88)`

> Which platform are you running on, what is

Windows 10 Version 20H2

**Comment by albert-github** [comment]
The `customdoxygen.css` you generated is most likely generated at the moment that there was no Doxyfile present or just an old Doxyfile without a `HTML_COLORSTYLE` setting, in this case the setting for the `HTML_COLORSTYLE` is `AUTO_LIGHT`.
When running the project with these settings and generating the chm you get the warning:
```
error: When enabling 'GENERATE_HTMLHELP' the 'HTML_COLORSTYLE' option should be either 'LIGHT' or 'DARK' but has value 'AUTO_LIGHT'. I'll adjust it for you to 'LIGHT'.
```
though the extra stylesheet is based on the setting `AUTO_LIGHT` and it has some settings in it that are not understood by the chm toolset (the features are a bit new for chm).

- I tried to generate the stylesheet for the `LIGHT` style but the `-w` command gave me an empty file (same happens with `DARK`), I'll look into this problem.

Workaround:
- take the  generated `doxygen.css` from the html directory and use this as base for your customdoxygen.css

**Comment by parfant** [comment]
Thank you for your support!

You are right. I generated the `customdoxygen.css` in a folder in which no doxyfile exists... This was my mistake.

When I generate the `customdoxygen.css` in the folder where the doxyfile is located, I also get an empty file.

The workaround with the generated `doxygen.css` works without any problems.

**Comment by albert-github** [comment]
The writing of the HTML style sheets,  in case of LIGHT / DARK color style, look like to have 2 problems:
- writing an empty file with the official 1.9.5 executable (though I cannot reproduce this with an own build executable)
- the problem that in case of LIGHT / DARK color style the values are not loaded yet, for this I've just pushed a proposed patch, pull request #9548

**Comment by lingr7** [comment]
> > How did you generate the `customdoxygen.css`?
> 
> I generated the customdoxygen.css in the command line with `doxygen -w html header.html footer.html customdoxygen.css`
> 
> > I assume you are running the released 1.9.5 version but to be sure can you give the full information provided by `doxygen -v`
> 
> `1.9.5 (2f6875a5ca481a69a6f32650c77a667f87d25e88)`
> 
> > Which platform are you running on, what is
> 
> Windows 10 Version 20H2

please forgive me for thinking about other issues, I want to know how to install chm generator , and  can't find the official download and installation package link from Microsoft.it is hard to find it.

**Comment by albert-github** [comment]
The problem with the chm / html help generator (`hhc.exe`) is has been abandoned by Microsoft a long time  ago (see also https://en.wikipedia.org/wiki/Microsoft_Compiled_HTML_Help). At first the distribution package was still on the Microsoft site but has been removed there as well some time ago. 

You might be able to find an old distribution somewhere on the internet in one or another archive.

**Comment by lingr7** [comment]
> The problem with the chm / html help generator (`hhc.exe`) is has been abandoned by Microsoft a long time ago (see also https://en.wikipedia.org/wiki/Microsoft_Compiled_HTML_Help). At first the distribution package was still on the Microsoft site but has been removed there as well some time ago.
> 
> You might be able to find an old distribution somewhere on the internet in one or another archive.

i find a method but must can not use when UAC limit.https://community.chocolatey.org/packages/html-help-workshop

**Comment by parfant** [comment]
> > > How did you generate the `customdoxygen.css`?
> > 
> > 
> > I generated the customdoxygen.css in the command line with `doxygen -w html header.html footer.html customdoxygen.css`
> > > I assume you are running the released 1.9.5 version but to be sure can you give the full information provided by `doxygen -v`
> > 
> > 
> > `1.9.5 (2f6875a5ca481a69a6f32650c77a667f87d25e88)`
> > > Which platform are you running on, what is
> > 
> > 
> > Windows 10 Version 20H2
> 
> please forgive me for thinking about other issues, I want to know how to install chm generator , and can't find the official download and installation package link from Microsoft.it is hard to find it.

You can find it here: https://github.com/EWSoftware/SHFB/raw/master/ThirdPartyTools/htmlhelp.exe

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.6.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
